gtsam/slam/lago.cpp

```diff
@@ -211,42 +216,77 @@ static PredecessorMap<Key> findOdometricPath(
       minUnassigned = false;
     }
     if (key2 - key1 == 1) { // consecutive keys
-      tree.insert(key2, key1);
+      tree.emplace(key2, key1);
       if (key1 < minKey)
         minKey = key1;
     }
   }
-  tree.insert(minKey, initialize::kAnchorKey);
-  tree.insert(initialize::kAnchorKey, initialize::kAnchorKey); // root
+  tree.emplace(minKey, kAnchorKey);
+  tree.emplace(kAnchorKey, kAnchorKey); // root
   return tree;
 }
 
+/*****************************************************************************/
+PredecessorMap findMinimumSpanningTree(
+    const NonlinearFactorGraph& pose2Graph) {
+  // Compute the minimum spanning tree
+  const auto edgeWeights = std::vector<double>(pose2Graph.size(), 1.0);
+  const auto mstEdgeIndices =
+      utils::kruskal(pose2Graph, edgeWeights);
+
+  // Create a PredecessorMap 'predecessorMap' such that:
+  // predecessorMap[key2] = key1, where key1 is the 'parent' node for key2 in
+  // the spanning tree
+  PredecessorMap predecessorMap;
```

Comment:
I guess you opted to not add to kruskal?

Reply:
Yes, for the following 2 reasons:
1. I think this function is a bit too specific for Lago and would require a bit more effort to make it more generic.
2. Making `kruskal` return a `PredecessorMap` would mean that we would have to re-extract the edge indices in `SubGraphBuilder` or make `kruskal` return both the `PredecessorMap` and the edge indices.